de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/invoice/process/CreateAdjustmentChargeFromInvoice.java

```diff
@@ -57,7 +58,8 @@ protected void prepare()
 			}
 			else if (CreateAdjustmentChargeFromInvoice.PARA_DocSubType.equals(name))
 			{
-				parameterDocSubType = (BigDecimal)parameter;
+				// convert the given value of the parameter to BigDecimal
+				parameterDocSubType = NumberUtils.asBigDecimal(parameter, new BigDecimal(-1));
```

Comment:
> Why is parameterDocType a number to start with?
The parameters has AD_Reference = Table. 

>  If the docSubtype parameter isn't actually needed (anymore), then please remove it and clean up the API.
The parameter is needed because the user can choose between 2 doc sub types when creating an adjustment charge ( like with the credit memo, but there the parameter has Reference = Table Direct)

> see what those AD_Table_ID and Record_ID fields are about 
They are needed for the PostProcess, for zooming into the newly created document